proposals/nnnn-ios-show-manager.md

```diff
@@ -0,0 +1,308 @@
+
+/**
+ Create a single-state soft button. For example, a button that brings up a Perform Interaction menu.
+
+ @param name The name of the button
+ @param eventHandler The handler to be called when the button is in the current state and is pressed
+ @param state The single state of the button
+ */
+- (void)initWithName:(NSString *)name state:(SDLSoftButtonState *)state handler:(SDLRPCButtonNotificationHandler)eventHandler;
+
+/**
+ Transition the soft button to another state in the `states` property. The wrapper considers all transitions valid (assuming a state with that name exists).
+ 
+ @warning This method will throw an exception and crash your app (on purpose) if you attempt an invalid transition. So...don't do that.
+
+ @param stateName The next state.
+ */
+- (void)transitionToState:(NSString *)stateName;
+
+@end
+
+NS_ASSUME_NONNULL_END
+```
+
+A wrapper of one or multiple states for a soft button
+
+##### SDLSoftButtonManager
+```objc
+NS_ASSUME_NONNULL_BEGIN
+
+@interface SDLSoftButtonManager : NSObject
+
+/**
+ An array of soft button wrappers containing soft button states. This is the current set of soft buttons in the process of being displayed or that are currently displayed. To display a new set of soft buttons, set this property.
+ */
+@property (copy, nonatomic) NSArray<SDLSoftButtonWrapper *> *softButtons;
+
+/**
+ Create a soft button manager with a connection manager
+
+ @param connectionManager The manager that forwards RPCs
+ @return A new instance of a soft button manager
+ */
+- (instancetype)initWithConnectionManager:(id<SDLConnectionManagerType>)connectionManager;
+- 
+/**
+ Update a particular button's current state into a new state and update the screen.
+
+ @param connectionManager The manager that forwards RPCs
+ @return A new instance of a soft button manager
+ */
+- (void)updateButtonNamed:(NSString *)buttonName replacingCurrentStateWithState:(SDLButtonState *)state;
+
+@end
+
+NS_ASSUME_NONNULL_END
+```
+
+The actual manager that handles translating the soft button wrappers and states into RPCs.
+
+#### Internals
+The soft button manager attempts to make it clearer and easier to set soft buttons, especially those that change themselves with different states when pressed, such as repeat or shuffle buttons. Developers create states for a specific button (though some buttons may need only one state), which is then composed into a button wrapper (because our naming sucks here), which is then set into the manager.
+
+The wrapper carries the event handler so that developers can transition states within the handler.
+
+To update the buttons, either developers can update an individual button to a new state through the `SDLSoftButtonWrapper`. If the buttons themselves are changing, the developer can set the `softButtons` array of the `SDLSoftButtonManager`.
```

Comment:
Should there be a second option added to the first sentence "either developers can update...or developers can..."?  If not, please remove "either."